docs/sql-migration-guide-upgrade.md

```diff
@@ -23,6 +23,8 @@ license: |
 {:toc}
 
 ## Upgrading From Spark SQL 2.4 to 3.0
+  - Since Spark 3.0, trim the string when cast string type to Boolean/Datetime/Numeric types.
```

Comment:
Could you complete this sentence by having a subject?
cc @srowen

Reply:
Yes something like "when casting from string to boolean, date or numeric types, whitespace is trimmed from the ends of the value first"

Reply:
Updated.